### ISSUE TITLE
When switching from joint mode to axis mode, linuxcnc warns outside limits when on boundary

### ISSUE THREAD

**Issue by ftkalcevic** [LinuxCNC/linuxcnc#656]
If when switching from joint mode to axis mode, and the axis is on the boundary, a warning is logged that you are starting outside limits, when you really aren't.

This is a bug in emc/motion/control.c update_teleop_with_check().
There are two comparisons....

```
 if  ( (axis->ext_offset_tp.curr_pos + axis->teleop_tp.curr_pos)
          >= axis->max_pos_limit) {
        // positive error, restore save_curr_pos
```

and

```
    if  ( (axis->ext_offset_tp.curr_pos + axis->teleop_tp.curr_pos)
           <= axis->min_pos_limit) {
        // negative error, restore save_curr_pos
```

which should use > and <, not >= and <=

(Sorry - my github fork is broken, otherwise I would have submitted a pull request)

**Comment by rene-dev** [comment]
I have noticed this as well, but why was this not an issue before?

**Comment by ftkalcevic** [comment]
> I have noticed this as well, but why was this not an issue before?

It's only a warning, so I'm guessing it was just ignored.

**Comment by rene-dev** [comment]
no, Im running master on all my machines, and suddenly I noticed getting those messages.

**Comment by andypugh** [comment]
Is anyone looking at this? Perhaps a git bisect would help see what changed? Or maybe just make the suggested fix?
